lib/scrabble-project/scrabble-class.rb

```diff
@@ -0,0 +1,90 @@
+module Scrabble
+
+  class Scrabble
+    # Set constants with general error message, point values of each letter, all letters
+    WORD_ERROR = "ERROR: Invalid input. Either your word has more that 7 letters or you entered something ridiculous. You only have 7 tiles, and Scrabble uses WORDS!"
+    POINTS = {
+      1 => ["a", "e", "i", "o", "u", "l", "n", "r", "s", "t"],
+      2 => ["d", "g"],
+      3 => ["b", "c", "m", "p"],
+      4 => ["f", "h", "v", "w", "y"],
+      5 => ["k"],
+      6 => ["j", "x"],
+      7 => ["q", "z"]
+    }
+
+    LETTERS = ("a".."z").to_a
+
+    def self.get_points(word)
+      # set initial value of score
+      score = 0
+
+      # convert to string, make letters lower case, & push letters to array
+      word_array = word.to_s.downcase.split("").push
+
+      # for each letter in word_array, check POINTS and add to total to create score
+      word_array.each do |letter|
+        for num in 1..7
+          if POINTS[num].include?(letter)
+            score += num
+          end
+        end
+      end
+      return score
+    end
+
+    # Check input to guard against empty, nil, or otherwise invalid parameter
+    # Word length cannot exceed 7
+    def self.valid_input?(word)
+      word = word.to_s.downcase
+
+      if (word == "") || (word == nil) || (word.length > 7) || (word.split("").all? { |letter| !LETTERS.include?(letter) })
+        valid = false
+      else
+        valid = true
+      end
+    end
+
+    def self.score(word)
+      # error if invalid input
+      if self.valid_input?(word) == false
+        WORD_ERROR
+
+      # run get_points if valid input
+      else
+        self.get_points(word)
+      end
+    end
+
+    def self.highest(array_of_words)
+
+      # score words and group by score value.
+      grouped_hash = array_of_words.group_by do |word|
+          score(word) end
+
```

Comment:
Style nit: Usually, the `end` should go on its own line after the method. If your block is short, try using curly braces instead (e.g. `{ |word| score(word) }` )